tests/test_array.py

```
@@ -0,0 +1,170 @@
+        """Mid-operation timeout should be wrapped in ToolError."""
+
+        _mock_graphql.side_effect = TimeoutError("operation timed out")
+        tool_fn = _make_tool()
+        with pytest.raises(ToolError, match="timed out"):
+            await tool_fn(action="shutdown", confirm=True)
+
+
+class TestArrayNetworkErrors:
+    """Tests for network-level failures in array operations."""
+
+    async def test_http_500_server_error(self, _mock_graphql: AsyncMock) -> None:
```

Ruff (0.15.0)

[warning] Fixture `_mock_graphql` without value is injected as parameter, use `@pytest.mark.usefixtures` instead

(PT019)